### PR TITLE
test: add tests for injector

### PR DESCRIPTION
**Pull request by phillebaba**
## Description

Adds tests to injector.

## Related Issue

Relates to #2512 

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/cluster/injector.go [hunk 1]
```diff
@@ -473,10 +473,8 @@ func (c *Cluster) getImagesAndNodesForInjection(ctx context.Context) (imageNodeM
 					continue
 				}
 
-				for _, taint := range nodeDetails.Spec.Taints {
-					if taint.Effect == corev1.TaintEffectNoSchedule || taint.Effect == corev1.TaintEffectNoExecute {
-						continue
```
Comment (phillebaba) [inline src/pkg/cluster/injector.go:478]: This continue does not skip the node, which is why I implemented the check as a function instead.

**Comment by phillebaba** [comment]
Injector will probably need some refactoring to be able to properly test the full logic.

File: src/pkg/cluster/injector_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+	cs := fake.NewSimpleClientset()
+	c := &Cluster{
+		&k8s.K8s{
+			Clientset: cs,
+		},
+	}
+
+	ctx := context.Background()
+	for i := 0; i < 2; i++ {
```
Comment (lucasrod16) [inline src/pkg/cluster/injector_test.go:41]: are these iterations there to test idempotence?

**Review by lucasrod16 — APPROVED** [review]
lgtm

**Review by Noxsios — APPROVED** [review]

**Review by AustinAbro321 — APPROVED** [review]

**Pull request merged**